operators/test/e2e/cmd/run/run.go

```diff
@@ -0,0 +1,462 @@
+	runFlags
+	eventLog       string
+	kubectlWrapper *command.Kubectl
+	testContext    test.Context
+	scratchDir     string
+	cleanupFuncs   []func()
+}
+
+func (h *helper) createScratchDir() error {
+	h.scratchDir = filepath.Join(h.scratchDirRoot, h.testRunName)
+	log.Info("Creating test output directory", "directory", h.scratchDir)
+
+	// ensure that the directory does not exist
+	if _, err := os.Stat(h.scratchDir); !os.IsNotExist(err) {
+		return errors.Wrapf(err, "scratch directory already exists: %s", h.scratchDir)
+	}
+
+	// create the directory
+	if err := os.MkdirAll(h.scratchDir, os.ModePerm); err != nil {
+		return errors.Wrapf(err, "failed to create scratch directory: %s", h.scratchDir)
+	}
+
+	// generate the path to the event log
+	h.eventLog = filepath.Join(h.scratchDir, "event.log")
+
+	// clean up the directory
+	h.addCleanupFunc(func() {
+		log.Info("Cleaning up the scratch directory", "directory", h.scratchDir)
+		if err := os.RemoveAll(h.scratchDir); err != nil {
+			log.Error(err, "Failed to cleanup scratch directory", "path", h.scratchDir)
+		}
+	})
+
+	return nil
+}
+
+func (h *helper) initTestContext() error {
+	h.testContext = test.Context{
+		AutoPortForwarding:  h.autoPortForwarding,
+		E2EImage:            h.e2eImage,
+		E2ENamespace:        h.testRunName,
+		E2EServiceAccount:   h.testRunName,
+		ElasticStackVersion: h.elasticStackVersion,
+		GlobalOperator: test.ClusterResource{
+			Name:      fmt.Sprintf("%s-global-operator", h.testRunName),
+			Namespace: fmt.Sprintf("%s-elastic-system", h.testRunName),
+		},
+		Local:              h.local,
+		NamespaceOperators: make([]test.NamespaceOperator, len(h.managedNamespaces)),
+		OperatorImage:      h.operatorImage,
+		TestLicence:        h.testLicence,
+		TestRegex:          h.testRegex,
+		TestRun:            h.testRunName,
+	}
+
+	for i, ns := range h.managedNamespaces {
+		h.testContext.NamespaceOperators[i] = test.NamespaceOperator{
+			ClusterResource: test.ClusterResource{
+				Name:      fmt.Sprintf("%s-%s-ns-operator", h.testRunName, ns),
+				Namespace: fmt.Sprintf("%s-ns-operators", h.testRunName),
+			},
+			ManagedNamespace: fmt.Sprintf("%s-%s", h.testRunName, ns),
+		}
+	}
+
+	// write the test context if required
+	if h.testContextOutPath != "" {
+		log.Info("Writing test context", "path", h.testContextOutPath)
+		f, err := os.Create(h.testContextOutPath)
+		if err != nil {
+			return errors.Wrap(err, "failed to write test context")
+		}
+
+		defer f.Close()
+		enc := json.NewEncoder(f)
+		if err := enc.Encode(h.testContext); err != nil {
+			return errors.Wrap(err, "failed to encode test context")
+		}
+	}
+
+	return nil
+}
+
+func (h *helper) createE2ENamespaceAndRoleBindings() error {
+	log.Info("Creating E2E namespace and role bindings")
+	return h.kubectlApplyTemplate("config/e2e/rbac.yaml", h.testContext, true)
+}
+
+func (h *helper) installCRDs() error {
+	log.Info("Installing CRDs")
+	crds, err := filepath.Glob("config/crds/*.yaml")
```

Comment:
I noticed that when running `go run cmd/main.go run --local`, there is no checking that we are on the right directory to find the yaml manifests.
Without `--local` it fails fast enough with `Error: failed to parse template at config/e2e/rbac.yaml: open config/e2e/rbac.yaml: no such file or directory`.

Reply:
Shoud we add a sanity check that we are in the right directory in all the modes?